src/Xrefcheck/Verify.hs

```diff
@@ -723,15 +751,18 @@ checkExternalResource Config{..} link
 
       let maxTime = Time @Second $ unTime ncExternalRefCheckTimeout * timeoutFrac
 
-      mres <- liftIO (timeout maxTime $ void reqLink) `catch`
-        (either throwError (\() -> return (Just ())) . interpretErrors)
-      maybe (throwError $ ExternalResourceSomeError "Response timeout") pure mres
-
-    isTemporaryRedirectCode :: Int -> Bool
-    isTemporaryRedirectCode = flip elem [302, 303, 307]
+      reqRes <- catch (liftIO (timeout maxTime $ reqLink $> RRDone)) $ \httpErr ->
```

Comment:
Now, instead of `Maybe ()`, the result is `Maybe ResponseResult`, where `Nothing` means timeout and `Just ResponseResult` tells "Done" or "Follow this redirect".